lib/puppetserver/ca/action/prune.rb

```diff
@@ -50,15 +50,22 @@ def run(inputs)
           loader = X509Loader.new(puppet.settings[:cacert], puppet.settings[:cakey], puppet.settings[:cacrl])
 
           puppet_crl = loader.crls.select { |crl| crl.verify(loader.key) }
-          prune_CRLs(puppet_crl)
-          update_pruned_CRL(puppet_crl, loader.key)
-          FileSystem.write_file(puppet.settings[:cacrl], loader.crls, 0644)
+          update_flag = prune_CRLs(puppet_crl)
+
+          if update_flag == true
+            update_pruned_CRL(puppet_crl, loader.key)
+            FileSystem.write_file(puppet.settings[:cacrl], loader.crls, 0640)
+            @logger.inform("Finished pruning Puppet's CRL.")
+          else
+            @logger.inform("No pruning needed.")
```

Comment:
This might be a little vague without context. Maybe something like
```suggestion
            @logger.inform("No duplicate revocations found in Puppet's CRL.")
```